back/cmd/openbsrv/db.go

```diff
@@ -19,14 +19,34 @@ func newSQLDB(driver, creds string) (*sql.DB, error) {
 	db.SetMaxIdleConns(128)
 	db.SetConnMaxLifetime(time.Hour)
 
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
-	defer cancel()
+	return db, patientPing(db)
+}
 
-	if err = db.PingContext(ctx); err != nil {
-		return nil, err
+func patientPing(db *sql.DB) error {
+	limit := time.Second * 3
+	pause := time.Millisecond * 500
```

Comment:
Is there a reason this can't be `pause = time.Second * 0.5`? Likely doesn't matter, but I'm just curious.

Reply:
Nope, no reason. I just picked a high enough time that it should be likely that something has gone wrong if the request hangs for that long. I believe that it should cycle immediately if there is no connection whatsoever, but I might be mistaken.

Reply:
oops, I had just responded without reading thoroughly. I'm quite sleepy...

time.Second is of type [time.Duration](https://golang.org/pkg/time/#Duration) which is a typed int64 . 0.5 would be an untyped float constant and cannot be mixed with an integer of any type. 500 is an untyped integer constant, so it can be mixed with the time.Duration type. That is, 500 becomes a time.Duration based on the context of it's usage.

https://play.golang.org/p/qWut6Eu4LJ7

If you'd like, you can read more here: https://blog.golang.org/constants

I hope I stated all of this clearly.